File: pec-observation-app/android-app/app/src/main/kotlin/br/gov/educacao/sp/pecobservation/ports/SyncQueuePort.kt
package br.gov.educacao.sp.pecobservation.ports

import br.gov.educacao.sp.pecobservation.domain.entities.SyncCommand
import java.util.UUID

interface SyncQueuePort {
    suspend fun enqueue(command: SyncCommand)
    suspend fun pendingCommands(): List<SyncCommand>
    suspend fun markInProgress(commandId: UUID)
    suspend fun markCompleted(commandId: UUID)
    suspend fun markFailed(commandId: UUID, reason: String)
    suspend fun pendingCount(): Int
}

File: pec-observation-app/ios-app/Sources/PECObservation/Ports/SyncQueuePort.swift
import Foundation

public protocol SyncQueuePort {
    func enqueue(_ command: SyncCommand) throws
    func pendingCommands() throws -> [SyncCommand]
    func markInProgress(_ commandId: UUID) throws
    func markCompleted(_ commandId: UUID) throws
    func markFailed(_ commandId: UUID, reason: String) throws
    func pendingCount() throws -> Int
}
